### PR TITLE
Use service instead of deprecated plist_options

### PR DESCRIPTION
**Pull request by thadc23**
Fixing this warning message:

```
Warning: Calling plist_options is deprecated! Use service.require_root instead.
Please report this issue to the cockroachdb/tap tap (not Homebrew/brew or Homebrew/homebrew-core), or even better, submit a PR to fix it:
  /usr/local/Homebrew/Library/Taps/cockroachdb/homebrew-tap/Formula/cockroach.rb:62
```

### CODE REVIEW

**Comment by rail** [comment]
oops, just realized that you submitted this PR :)
I created https://github.com/cockroachdb/homebrew-tap/pull/136 which contains [this condition](https://github.com/cockroachdb/homebrew-tap/pull/136/files#diff-10d1c12c97dc80980ba3b97e51d4bae24ee4e9a4690a63844ddfa2efdc172362R70-R72) for M1 Macs. Can you incorporate it please.

File: Formula/cockroach.rb [hunk 1]
```diff
@@ -59,36 +59,12 @@ def caveats; <<~EOS
...
-      <string>#{var}</string>
-      <key>RunAtLoad</key>
-      <true/>
-      <key>KeepAlive</key>
-      <true/>
-    </dict>
-    </plist>
-  EOS
+  service do
+    run [bin/"cockroach", "start-single-node", "--insecure", "--http-port=26256", "--host=localhost"]
```
Comment (rail) [inline Formula/cockroach.rb:63]: I think this should be `opt_bin`

**Review by rail — CHANGES_REQUESTED** [review]
Can you also update https://github.com/cockroachdb/homebrew-tap/blob/cefa6e7254d0fb1755cd06ade30b152653accf85/release/cockroach-tmpl.rb#L62

**Comment by thadc23** [comment]
Working on these.  Thanks!

**Comment by thadc23** [comment]
@rail Should be incorporated now.

**Review by rail — APPROVED** [review]
TYVM!

**Pull request merged**